frontend/src/components/organisms/EventCancelForm.tsx

```diff
@@ -10,159 +10,201 @@ import MultilineTextField from "../atoms/MultilineTextField";
 import { Typography, Grid } from "@mui/material";
 import { useRouter } from "next/router";
+import { BASE_URL } from "@/utils/constants";
+import { auth } from "@/utils/firebase";
+import { useAuth } from "@/utils/AuthContext";
+import { fetchUserIdFromDatabase } from "@/utils/helpers";
+
+type eventData = {
+	eventid: string;
+	location: string;
+	datetime: string;
+	supervisors: string[];
+	capacity: number;
+	image_src: string;
+	tags: string[] | undefined;
+};
 
 interface EventCancelFormProps {
-  eventid: string;
-  location: string;
-  datetime: string;
-  supervisors: string[];
-  capacity: number;
-  image_src: string;
-  tags?: string[];
+	eventDetails: eventData;
 }
+
 type FormValues = {
-  cancelReason: string;
+	cancelReason: string;
 };
+const url = BASE_URL as string;
 
 /**
  * A confirmation modal body
  */
 const ModalBody = ({
-  eventid,
-  handleClose,
+	eventid,
+	handleClose,
+	cancelationMessage,
 }: {
-  eventid: string;
-  handleClose: () => void;
+	eventid: string;
+	handleClose: () => void;
+	cancelationMessage: string;
 }) => {
-  const router = useRouter();
-  const cancel = () => {
-    router.replace(`/events/${eventid}/confirm/cancel`);
-  };
-  return (
-    <div>
-      <Typography align="center" sx={{ paddingBottom: 2 }}>
-        Are you sure you want to cancel?
-      </Typography>
-      <Grid container spacing={2}>
-        <Grid item md={6} xs={12}>
-          <Button color="gray" type="button" onClick={handleClose}>
-            No
-          </Button>
-        </Grid>
-        <Grid item md={6} xs={12}>
-          <Button color="dark-gray" type="button" onClick={cancel}>
-            Yes, cancel
-          </Button>
-        </Grid>
-      </Grid>
-    </div>
-  );
+	const router = useRouter();
+	const { user } = useAuth();
+
+	const cancel = async () => {
+		const userToken = await auth.currentUser?.getIdToken();
+		const attendeeid = await fetchUserIdFromDatabase(
+			user?.email as string,
+			userToken as string
+		);
+
+		const fetchUrl = `${url}/events/${eventid}/attendees`;
+
+		const cancellationData = {
+			attendeeid: attendeeid,
+			cancelationMessage: cancelationMessage,
+		};
+		try {
+			const response = await fetch(fetchUrl, {
+				method: "PUT",
+				headers: {
+					Authorization: `Bearer ${userToken}`,
+					"Content-Type": "application/json",
+				},
+				body: JSON.stringify(cancellationData),
+			});
+			// Response Management
+			if (response.ok) {
+			} else {
+			}
+		} catch (error) {
+		}
+
+		router.reload();
+	};
+
+	return (
+		<div>
+			<Typography align="center" sx={{ paddingBottom: 2 }}>
+				Are you sure you want to cancel?
+			</Typography>
+			<Grid container spacing={2}>
+				<Grid item md={6} xs={12}>
+					<Button color="gray" type="button" onClick={handleClose}>
+						No
+					</Button>
+				</Grid>
+				<Grid item md={6} xs={12}>
+					<Button color="dark-gray" type="button" onClick={cancel}>
+						Yes, cancel
+					</Button>
+				</Grid>
```

Comment:
<img width = "500" src = "https://github.com/cornellh4i/lagos-volunteers/assets/44273288/7da79a8a-6e50-4339-9b85-b88fe9db6270" />

Same problem here with the buttons appearing blue instead of grey, might be an issue with modal or button.

Reply:
Issue with Modal!